File: 0x09-bfs/15_13549/13549.py
import sys
from collections import deque

input = sys.stdin.readline

n, k = map(int, input().split())

# dist[i] : i번째 위치까지 도달하는 데 걸리는 최소 시간
MAX_SIZE = 100001
dist = [-1] * MAX_SIZE
dist[n] = 0

queue = deque()
queue.append(n)

while queue:
    x = queue.popleft()
    if x == k: break
    # 1. 순간이동
    next = x*2
    if 0 <= next < MAX_SIZE and dist[next] == -1:
        dist[next] = dist[x]
        queue.appendleft(next)
    # 2. 걷기
    if x-1 >= 0 and dist[x-1] == -1:
        dist[x-1] = dist[x]+1
        queue.append(x-1)
    if x+1 < MAX_SIZE and dist[x+1] == -1:
        dist[x+1] = dist[x] +1
        queue.append(x+1)
print(dist[k])

File: 0x09-bfs/15_13549/13549.java
import java.io.*;
import java.util.*;

public class Main {
    static int n, m;
    static int[] dist;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        n = Integer.parseInt(st.nextToken());
        m = Integer.parseInt(st.nextToken());
        dist = new int[100001];

        Arrays.fill(dist, -1);

        Deque<Integer> deque = new ArrayDeque<>();
        deque.add(n);
        dist[n] = 0;

        while (!deque.isEmpty()) {
            int current = deque.poll();

            if (current == m) break;

            // 순간이동: 비용 0 → 앞에 넣기
            int teleport = current * 2;
            if (teleport <= 100000 && dist[teleport] == -1) {
                dist[teleport] = dist[current];
                deque.addFirst(teleport);
            }

            // 걷기: 비용 1 → 뒤에 넣기
            for (int next : new int[]{current - 1, current + 1}) {
                if (next >= 0 && next <= 100000 && dist[next] == -1) {
                    dist[next] = dist[current] + 1;
                    deque.addLast(next);
                }
            }
        }

        System.out.println(dist[m]);
    }
}
